README.md

```diff
@@ -214,4 +214,210 @@ internal sealed class AccountTypeRepository : IAccountTypeRepository
+
+## Data Transfer Object Persistence
+
+In most applications, it is desirable to completely separate persistence concerns from domain logic. The above examples slightly violate this principle by leaking some Json serialization details into the domain objects. However, this can be easily avoided, at the expense of having to define some Data Transfer Objects (DTOs), and mapping functions. Consider essentially the same domain objects, but with all serialization logic removed.
+
+```csharp
+public struct AccountId : IEquatable<AccountId>
+{
+    public static explicit operator string(AccountId id) { ... }
+
+    public static explicit operator AccountId(string value) { ... }
+
+    ...
+}
+
+...
+
+public struct TransactionId : IEquatable<TransactionId>
+{
+    public static explicit operator string(TransactionId id) { ... }
+
+    public static explicit operator TransactionId(string value) { ... }
+
+    ...
+}
+
+...
+
+public struct AccountType : IEquatable<AccountType>
+{
+    public AccountType(
+        string name,
+        decimal rate,
+        ...)
+    {
+        Name = name;
+        Rate = rate;
+        ...
+    }
+
+    public string Name { get; }
+
+    public decimal Rate { get; }
+
+    ...
+}
+```
+
+The persistence layer DTOs, including serialization logic, for these simple types would then look very similar to the original domain objects.
+
+```csharp
+internal struct AccountDto
+{
+    [JsonConstructor]
+    public AccountDto(
+        string id,
+        AccountTypeDto type,
+        ...)
+    {
+        Type = type;
+        ...
+    }
+
+    public string Id { get; }
+
+    public AccountTypeDto Type { get; }
+
+    ...
+}
+
+internal struct TransactionDto
+{
+    [JsonConstructor]
+    public TransactionDto(
+        string id,
+        string sender,
+        string recipient,
+        ...)
+    {
+        Sender = sender;
+        Recipient = recipient;
+        ...
+    }
+
+    public string Id { get; }
+
+    public string Sender { get; }
+
+    public string Recipient { get; }
+
+    ...
+}
+
+internal struct AccountTypeDto
+{
+    [JsonConstructor]
+    public AccountTypeDto(
+        string name,
+        decimal rate,
+        ...)
+    {
+        Name = name;
+        Rate = rate;
+        ...
+    }
+
+    public string Name { get; }
+
+    public decimal Rate { get; }
+
+    ...
+}
+```
+
+The overloaded implementations of both `IEntityFacade` and `IValueObjectFacade` can be created from the same factories, but by specifying the appropriate DTO and mapping function. They can be manually constructed as
+
+```csharp
+IEntityFacade<Account, AccountId, AccountDto> accountFacade = entityFacadeFactory.Create<Account, AccountId, AccountDto>(
+    database,
+    documentCollection,
+    a => new AccountDto((string)a.Id, new AccountTypeDto(a.Type.Name, a.Type.Rate)),
```

Comment:
I think it would be nice to put methods onto the DTO types called `ToDomainObject` and `FromDomainObject` that does the mapping logic. It will aid the readability of these lamdbas and keep the mapping logic encapsulated in the DTO.